Server/src/main/java/org/openas2/lib/helper/BCCryptoHelper.java

```diff
@@ -396,10 +386,25 @@ public MimeBodyPart verifySignature(MimeBodyPart part, Certificate cert) throws
                     logger.trace("Signer Attributes: data not available.");
                 }
             }
-            if (signer.verify(signerInfoVerifier)) {
-                logSignerInfo("Verified signature for signer info", signer, part, x509Cert);
-                return signedPart.getContent();
+
+            // Claudio Degioanni claudio.degioanni@bmeweb.it 05/11/2021
+            try {
+                if (signer.verify(signerInfoVerifier)) {
+                    logSignerInfo("Verified signature for signer info", signer, part, x509Cert);
+                    return signedPart.getContent();
+                }
+            } catch (CMSVerifierCertificateNotValidException ex) {
+                String as2SignIgnoreTimeIssue = Properties.getProperty("as2_sign_allow_expired_certificate", "false");
+
+                if ("true".equalsIgnoreCase(as2SignIgnoreTimeIssue)) {
+                    // if flag is enabled log only issue
+                    logSignerWarn("OUTDATED CERTIFICATE signer info", signer, part, x509Cert);
+                    return signedPart.getContent();
```

Comment:
The signing must still be verified even if the date stamp on the certificate is ignored so use this code to verify the signature inside the catch block ignoring the certificate date if the CMSVerifierCertificateNotValidException exception is thrown:
```
JcaSimpleSignerInfoVerifierBuilder verifierBuilder = new JcaSimpleSignerInfoVerifierBuilder();
signer.verify(verifierBuilder.build(x509Cert));
```

Reply:
Sorry i dont know very well the implementation the verify check certificare ad signing time ? 
To solve the code reported check certificate but not the exporation ? 

JcaSimpleSignerInfoVerifierBuilder verifierBuilder = new JcaSimpleSignerInfoVerifierBuilder();
signer.verify(verifierBuilder.build(x509Cert));